### DISCUSSION TITLE
Can openfast only calculate platform and mooring, ignoring the action of tower and blade

### DISCUSSION THREAD

**Discussion by shijieQiu** [OpenFAST/openfast#2146] [Q&A]
Hello everyone, may I ask if openfast can only calculate the platform and mooring, ignoring the action of tower and blade? If possible, could you show me the idea? Thank you

**Comment by jjonkman** [comment]
Dear @shijieQiu,

While OpenFAST doesn't have an option to not include the tower, nacelle, drivetrain, and rotor, you could avoid modeling these components by setting the input parameters such that their effect is negligible, such as by disabling the tower, nacelle yaw, drivetrain, and blade degrees of freedom in ElastoDyn and setting the lengths and masses/inertias to zero (or near zero for the parameters that are not allowed to be set exactly zero).

See a related question discussed here: https://github.com/OpenFAST/openfast/discussions/2040.

Best regards,

**Reply by shijieQiu** [reply]
Dear @jjonkman,
Thank you for your reply. I have modified according to the idea you gave. Now I only use three modules ElastDyn, HydroDyn and SubDyn. When ElastDyn and HydroDyn are used together, no error message is displayed. However, when ElastDyn, HydroDyn, and SubDyn are used together, error messages are displayed as follows:
FAST_Solution:FAST_AdvanceStates:ED_ABM4:ED_AB4:ED_RK4:ED_CalcContStateDeriv:SetCoordSy:Small
angle assumption violated in SUBROUTINE SmllRotTrans() due to a large platform displacement
(ElastoDyn SetCoordSy). The solution may be inaccurate. Simulation continuing, but future
warnings from SmllRotTrans() will be suppressed.
 Additional debugging message from SUBROUTINE SmllRotTrans(): 2.50000E-02 s
FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption1:FullOpt1_InputOutputSolve:SD_CalcOutput:
Angles in GetSmllRotAngs() are larger than 0.4 radians.
SD_CalcOutput: Angles in GetSmllRotAngs() are larger than 0.4 radians.
FullOpt1_InputOutputSolve:HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are
larger than 0.4 radians.
FullOpt1_InputOutputSolve:SD_CalcOutput: Angles in GetSmllRotAngs() are larger than 0.4 radians.
SD_CalcOutput: Angles in GetSmllRotAngs() are larger than 0.4 radians.
FullOpt1_InputOutputSolve:HydroDyn_CalcOutput:HDOut_MapOutputs: Angles in GetSmllRotAngs() are
larger than 0.

 OpenFAST encountered an error at simulation time 2.00000E-02 of 60 seconds.
 Simulation error level: FATAL ERROR

 Aborting OpenFAST.

I modified the model parameters and initial coordinates, and I suspect that the platform quality parameters were incorrectly defined. But after many attempts, I still couldn't solve the problem. I hope you can give me some advice.

Best regards,

**Reply by jjonkman** [reply]
Dear @shijieQiu,

It looks like your model is unstable, but it is difficult to know why without more information.

In the following OpenFAST Discussion, I provided guidance for running OpenFAST with structure-structure and hydro-elastic coupling between ElastoDyn, SubDyn, and HydroDyn: https://github.com/OpenFAST/openfast/discussions/801.  Are you following this guidance?

Best regards
